monitoring/router.nginx.conf

```diff
@@ -1,20 +1,45 @@
 server {
-    server_name _;
+    server_name grafana.*;
 
     location /monitoring/grafana/ {
         proxy_pass http://grafana/;
-        proxy_set_header Host internal.hail.is;
+        proxy_set_header Host $host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Host $host;
+        proxy_set_header X-Forwarded-Proto https;
     }
 
+    listen 80;
+    listen [::]:80;
+}
+
+server {
+    server_name prometheus.*;
+
     location /monitoring/prometheus/ {
-        proxy_pass http://prometheus;
-        proxy_set_header Host internal.hail.is;
+        proxy_pass http://prometheus/;
```

Comment:
You changed this to a trailing slash.  That means the location block now strips /monitor/prometheus which is probably why you're having trouble with prometheus.

Reply:
> Let me know if you want me to take this over.

I have a professional interest in seeing this through.

Reply:
> why you're having trouble with prometheus.

The behavior is totally unchanged from the configuration that had one monolithic server block. To figure out whether I had broken something, I restored John's config. Nothing changed, although it's possible I was seeing the effect of a crashloopbackoff say (though I checked that the pods were up each test, may have made a mistake).

The slash was added because the behavior looked inconsistent with Grafana's config. I read now that Prometheus expects the entire external url suffix, so what you say is consistent. We can add  "-web.route-prefix=/" to allow / to work.


> That means the location block now strips /monitor/prometheus

I thought so as well, but I wasn't seeing that behavior in logs. When I look at request logs of the grafana pod that received those requests, the request was /monitoring/grafana/ not /. That confused me, wasn't able to resolve last night.

I added the slash because 

I'll try again, and report the outcome.

Reply:
> The behavior is totally unchanged from the configuration that had one monolithic server block. 

 I found a clue that may have contributed. I just tried modifying the monitoring.yaml config to add a prometheus feature. Did it, had a typo, sent to crashloopbackoff. Updated the correct setting, deployed again, and the pod wasn't sent back to containercreating. Restored the original monitoring.yaml config, deployed again, again no change. I needed to manually kill the pod for it to pick up the config.